# Description of `Schools` table

In [1]:
import pandas as pd
import seaborn as sns
sns.set_theme()

In [3]:
df = pd.read_csv('../tables_creation/final_tables/schools_edrpou.csv')
df.head()

eoname  KOATUU_2020  \
0  Вiдокремлений структурний пiдроздiл "Краматорс...    510100000   
1  ВСП "Технологічно-промисловий фаховий коледж В...    510100000   
2             Вище професійне училище №11 м. Вінниці    510100000   
3  Вище художнє професійно-технічне училище № 5 м...    510100000   
4  Вище художнє професійно-технічне училище №5 м....    510100000   

                                   year    EDRPOU  
0                                [2023]  04601943  
1                                [2022]  00419667  
2  [2018, 2019, 2020, 2021, 2022, 2023]  03065891  
3                                [2023]  02539890  
4        [2018, 2019, 2020, 2021, 2022]  02539890

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32060 entries, 0 to 32059
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   eoname       32060 non-null  object
 1   KOATUU_2020  32060 non-null  int64 
 2   year         32060 non-null  object
 3   EDRPOU       32060 non-null  object
dtypes: int64(1), object(3)
memory usage: 1002.0+ KB


In [5]:
# print unique values for each column
for col in df.columns:
    print('Unique length in ', col+':\t', len(df[col].unique()))

Unique length in  eoname:	 31904
Unique length in  KOATUU_2020:	 7165
Unique length in  year:	 165
Unique length in  EDRPOU:	 12885


Features:
- eoname (Name of school): 31904 unique values of the school names which are used. It can be duplicates of schools due to mistakes in the data, better to use only EDRPOU.
- KOATUU_2020 (KOATUU code): 7165 unique codes (a unique digital code for each administrative-territorial entity of Ukraine, used from January 1998 till January 2021) of the schools territories. KOATUU codes in Ukraine follow a six-digit format. The first two digits signify the region, the next two represent the district, and the following two denote the city or municipality.
- year: lists of years in which certain school names (eoname) are used (one school can have different school names in different years). Range of values in lists:
    - 2016-2023 (list of integers)
- EDRPOU (unique code of the legal entity or physical person): 12885 unique codes of schools, which uniquely define the schools.